src/Http/Routing/src/ConfigureRouteHandlerOptions.cs

```diff
@@ -8,16 +8,16 @@ namespace Microsoft.AspNetCore.Routing;
 
 internal sealed class ConfigureRouteHandlerOptions : IConfigureOptions<RouteHandlerOptions>
 {
-    private readonly IHostEnvironment _environment;
+    private readonly IHostEnvironment? _environment;
 
-    public ConfigureRouteHandlerOptions(IHostEnvironment environment)
+    public ConfigureRouteHandlerOptions(IHostEnvironment? environment = null)
```

Comment:
What's the motivation for this change?

Reply:
It's to avoid breaking unit tests that add options to the service collection but not IHostEnvironment. There were a few tests in downstream projects like SignalR that were broken by this. This was a simple fix that might also avoid breaking external tests and odd configurations without an IHostEnvironment.